src/containers/Identity/PersonalInfo/ClaimManager/View.js

```diff
@@ -1,51 +1,71 @@
-import React, { useState, useEffect } from 'react';
-import { View, Text, Platform } from 'react-native';
+import React from 'react';
+import { Map as IMap } from 'immutable';
+import { View, Text } from 'react-native';
 import { ScrollView, TouchableOpacity } from 'react-native-gesture-handler';
 
 import { CheckBox } from 'react-native-elements';
-import data from './mockData';
 import styles from './styles';
 
-const ClaimManager = ({ navigation }) => {
-  const [claims, setClaims] = useState(data);
+const ClaimManager = (props) => {
+  const {
+    claims,
+    selectedClaims,
+    navigation,
+    claimCategories,
+    actions: {
+      setClaimVisibility,
+      updateSelectedClaims,
+      clearSelectedClaims,
+    },
+  } = props;
 
-  const toggleCheckbox = (id) => {
-    // const checkedClaim = claims.find((claim) => claim.id === id);
-    // checkedClaim.checked = !checkedClaim.checked;
-
-    // setClaims([...claims, checkedClaim]);
+  const toggleClaimVisibility = (selectedClaim) => {
+    if (selectedClaim.get('hidden')) {
+      setClaimVisibility(selectedClaim.get('id', ''), false);
+    } else {
+      setClaimVisibility(selectedClaim.get('id', ''), true);
+    }
   };
 
+  const moveSingleClaim = (selectedClaim) => {
+    clearSelectedClaims();
+    updateSelectedClaims(selectedClaim);
+    navigation.navigate('MoveIntoCategory');
+  };
 
-  const moveIn = () => {
+  const moveSelectedCategories = () => {
     navigation.navigate('MoveIntoCategory');
   };
 
+  const selectClaim = (selectedClaim) => {
+    updateSelectedClaims(selectedClaim);
+  };
+
+  const checkIfClaimIsSelected = (claim) => selectedClaims.indexOf(claim) > -1;
```

Comment:
These kind of lookups are slow. Avoid them. If you end up having them, be sure to use useMemo